sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -484,8 +521,11 @@ class SessionCatalog(
     val dbTables =
       externalCatalog.listTables(dbName, pattern).map { t => TableIdentifier(t, Some(dbName)) }
     synchronized {
-      val _tempTables = StringUtils.filterPattern(tempTables.keys.toSeq, pattern)
+      var _tempTables = StringUtils.filterPattern(tempTables.keys.toSeq, pattern)
         .map { t => TableIdentifier(t) }
+      if (db != INFORMATION_SCHEMA_DATABASE) {
```

Comment:
This part is not clear to me. 
What happens if users want to list tables in database `INFORMATION_SCHEMA_DATABASE`, it will return empty?